### PR TITLE
chore: don't wrap the indexes table header cells

### PR DESCRIPTION
**Pull request by lerouxb**
I couldn't look at this anymore

before:
Screenshot 2024-09-24 at 13 45 55

after:
Screenshot 2024-09-24 at 14 00 07

### CODE REVIEW

File: packages/compass-indexes/src/components/indexes-table/indexes-table.tsx [hunk 1]
```diff
@@ -126,7 +127,7 @@ export function IndexesTable<T>({
           isSticky
           className={cx(tableHeadStyles, darkMode && tableHeadDarkModeStyles)}
         >
-          {table.getHeaderGroups().map((headerGroup: HeaderGroup<T>) => (
+          {table.getHeaderGroups().map((headerGroup) => (
```
Comment (lerouxb) [inline packages/compass-indexes/src/components/indexes-table/indexes-table.tsx:130]: This is a drive-by to get rid of the typescript error.

**Review by Anemy — APPROVED** [review]

**Review by gribnoysup — APPROVED** [review]
